fields.go

```diff
@@ -0,0 +1,156 @@
+// Package environschema implements a way to specify
+// configuration attributes for Juju environments.
+package environschema // gopkg.in/juju/environschema.v1
+import (
+	"fmt"
+	"github.com/juju/schema"
+	"strings"
+)
+
+// What to do about reading content from paths?
+// Could just have a load of client-side special cases.
+
+// Fields holds a map from attribute name to
+// information about that attribute.
+type Fields map[string]Attr
+
+type Attr struct {
+	// Description holds a human-readable description
+	// of the attribute.
+	Description string `json:"description"`
+
+	// Type holds the type of the attribute value.
+	Type FieldType `json:"type"`
+
+	// Mandatory specifies whether the attribute
+	// must be specified. If Default is also specified,
+	// this field is redundant, as the attribute will be
+	// filled in with the default value if missing.
+	Mandatory bool `json:"mandatory,omitempty"`
+
+	// Secret specifies whether the attribute should be
+	// considered secret.
+	Secret bool `json:"is-secret,omitempty"`
+
+	// EnvVar holds the environment variable
+	// that will be used to obtain the default value
+	// if it isn't specified.
+	EnvVar string `json:"env-var,omitempty"`
+
+	// Default holds the default value of the attribute
+	// when not specified.
+	Default interface{} `json:"default,omitempty"`
+
+	// Example holds an example value for the attribute
+	// that can be used to produce a plausible-looking
+	// entry for the attribute without necessarily using
+	// it as a default value.
+	Example interface{} `json:"example,omitempty"`
+
+	// Values holds the set of all possible values of the attribute.
+	Values []interface{} `json:"values,omitempty"`
+}
+
+// FieldType describes the type of an attribute value.
+type FieldType string
+
+// The following constants are the possible type values.
+const (
+	Tstring FieldType = "string"
```

Comment:
I'd be more inclined to rename FieldType to Type than to use FT, I think, if you want that to be consistent. I'm reasonably happy with the T prefix, although I'm not entirely sure that it shouldn't be TString, TBool, etc.